packages/Confirmation/src/Confirmation.js

```diff
@@ -73,29 +75,27 @@ const Confirmation = props => {
     popoverKey = uuid();
   }, [isPending]);
 
-  const triggerProps = {
-    isConfirmOpen,
-    handleOpenConfirm,
-  };
-
   const handleCloseConfirm = () => {
-    if (isConfirmOpen) {
-      setIsConfirmOpen(false);
-      if (triggerRef.current) triggerRef.current.focus();
-      setTimeout(onClose, 250);
-    }
+    setIsConfirmOpen(prevIsConfirmOpen => {
+      if (prevIsConfirmOpen) {
+        if (triggerRef.current) triggerRef.current.focus();
+        setTimeout(onClose, 250);
```

Comment:
You dont need the `setTimeout` anymore i thought?

Reply:
the `setTimeout` was already existing and is a delay to allow for the animation of popover closing. It's not related to the check of `prevIsConfirmOpen`

Reply:
ohhh you right, there was a` todo` to add onAnimationEnd and onAnimationStart to the popover, can we create an issue for that? and refer to this conversation?

Reply:
Sure will add to popover issues

Reply:
nvm I created it https://github.com/acl-services/paprika/issues/352